### ISSUE TITLE
problem  for ctc

### ISSUE THREAD

**Issue by wWzZb** [woshisad159/TFNet#2]
It seems that ctcdecode cannot run on Windows. May I ask which system you are using? If you're using Windows, could you provide a solution?

**Comment by woshisad159** [comment]
I'm using linux and ctcdecode only works on linux, no other version.


------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "woshisad159/TFNet"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2024年11月21日(星期四) 晚上10:33
***@***.***&gt;;
***@***.***&gt;;
主题:&nbsp;[woshisad159/TFNet] problem  for ctc (Issue #2)





 
It seems that ctcdecode cannot run on Windows. May I ask which system you are using? If you're using Windows, could you provide a solution?

**Comment by wWzZb** [comment]
Thank you for your project. May I ask what GPU you used for your experiments? My GPU is RTX 4060 with 8GB of memory, but it seems the configuration might not be successful

![image](https://github.com/user-attachments/assets/2a347a5f-3b28-4f5d-a68d-d8bb389d2b24)

**Comment by woshisad159** [comment]
I use a 3090Ti GPU with 24GB of memory.


------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "woshisad159/TFNet"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2024年11月23日(星期六) 下午4:00
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [woshisad159/TFNet] problem  for ctc (Issue #2)





 
Thank you for your project. May I ask what GPU you used for your experiments? My GPU is RTX 4060 with 8GB of memory, but it seems the configuration might not be successful
 
image.png (view on web)

**Comment by wWzZb** [comment]
Why is there no file in the train folder saving the prediction results of the CE-CSL dataset, and only these two RWTH  in the images?

**Comment by wWzZb** [comment]
Why is there no file in the train folder saving the prediction results of the CE-CSL dataset, and only these two RWTH appear in the images?
![image](https://github.com/user-attachments/assets/e2283ad6-ef14-418a-a446-a43209a075bd)

**Comment by woshisad159** [comment]
I am using the pycharm compilation environment to run it and also print out the results of the run, the results of the run for the dataset CE-CSL are not saved, but the model parameters of the run are saved.


So you need to use the pycharm compilation environment to run and see the printouts


------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "woshisad159/TFNet"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2024年11月27日(星期三) 中午1:47
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [woshisad159/TFNet] problem  for ctc (Issue #2)





 
Why is there no file in the train folder saving the prediction results of the CE-CSL dataset, and only these two RWTH appear in the images?
 image.png (view on web)

**Comment by wWzZb** [comment]
Thank you for the response. May I ask how many epochs of training are needed to achieve the WER value on the test set as reported in the paper

**Comment by nanana-121** [comment]
Hello, have you successfully installed `ctcdecode`? My environment is PyTorch 1.11.0, CUDA 11.3, and Ubuntu 20.04. I have tried multiple times but still cannot install it successfully. Do you have any tips or successful experiences that you could share with me?

**Comment by woshisad159** [comment]
Yes, I have successfully installed ctcdecode. I know of an installation issue, when you download the ctcdecode code directly from github to install it, the third_party file at this point is incomplete, and you still need to download the relevant files.




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "woshisad159/TFNet"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2025年3月7日(星期五) 下午4:38
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [woshisad159/TFNet] problem  for ctc (Issue #2)



   
Hello, have you successfully installed ctcdecode? My environment is PyTorch 1.11.0, CUDA 11.3, and Ubuntu 20.04. I have tried multiple times but still cannot install it successfully. Do you have any tips or successful experiences that you could share with me?

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you commented.Message ID: ***@***.***&gt;
  nanana-121 left a comment (woshisad159/TFNet#2)
 
Hello, have you successfully installed ctcdecode? My environment is PyTorch 1.11.0, CUDA 11.3, and Ubuntu 20.04. I have tried multiple times but still cannot install it successfully. Do you have any tips or successful experiences that you could share with me?

**Comment by nanana-121** [comment]
Hello, I have solved the CTCdecode issue. Have you ever used the trained model for actual sign language video recognition? I used your trained model to recognize multiple sign language videos in the test file, but the results were not very ideal. It can recognize individual words in a sentence, but it is not yet able to recognize entire sentences.

**Comment by woshisad159** [comment]
Hi, I've run it on a test set with a trained model and the best WER I could get was only about 32%, which is a relatively low accuracy, but it's the best I've gotten so far. 
It needs our joint efforts to improve the accuracy.





------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "woshisad159/TFNet"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2025年3月13日(星期四) 下午2:27
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [woshisad159/TFNet] problem  for ctc (Issue #2)



   
Hello, I have solved the CTCdecode issue. Have you ever used the trained model for actual sign language video recognition? I used your trained model to recognize multiple sign language videos in the test file, but the results were not very ideal. It can recognize individual words in a sentence, but it is not yet able to recognize entire sentences.

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you commented.Message ID: ***@***.***&gt;
  nanana-121 left a comment (woshisad159/TFNet#2)
 
Hello, I have solved the CTCdecode issue. Have you ever used the trained model for actual sign language video recognition? I used your trained model to recognize multiple sign language videos in the test file, but the results were not very ideal. It can recognize individual words in a sentence, but it is not yet able to recognize entire sentences.

**Comment by wWzZb** [comment]
In the code downloaded from GitHub for ctcdecode, one of the two pieces of code under third_party seems to be missing.

![Image](https://github.com/user-attachments/assets/e3aebee4-da33-4e65-8b93-176f2b021d69)

You need to clone it from the corresponding project and place it in the appropriate directory.

This is my think of the issue, but I'm not entirely sure. You can refer to this as a reference.

**Comment by woshisad159** [comment]
You are right in thinking that the corresponding documentation is indeed missing.




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "woshisad159/TFNet"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2025年3月13日(星期四) 下午3:59
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [woshisad159/TFNet] problem  for ctc (Issue #2)



   
In the code downloaded from GitHub for ctcdecode, one of the two pieces of code under third_party seems to be missing.
 
image.png (view on web)
 
You need to clone it from the corresponding project and place it in the appropriate directory.
 
This is my think of the issue, but I'm not entirely sure. You can refer to this as a reference.

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you commented.Message ID: ***@***.***&gt;
  wWzZb left a comment (woshisad159/TFNet#2)
 
In the code downloaded from GitHub for ctcdecode, one of the two pieces of code under third_party seems to be missing.
 
image.png (view on web)
 
You need to clone it from the corresponding project and place it in the appropriate directory.
 
This is my think of the issue, but I'm not entirely sure. You can refer to this as a reference.
